File: Python/Program_Studi.py
# Mmebuat kelas Program Studi dengan hubungan composite ke kelas Course
class Program_Studi:
    # Membuat beberapa atirbut private
    # prodi Course
    __prodi = ""
    __kode = ""

    # Constructor with default initialization
    def __init__(self, prodi="", kode=""):
        self.__prodi = prodi
        self.__kode = kode
        self.course_list = []

    # Setter and Getter methods for private attributes
    # set prodi
    def setProdi(self, prodi):
        self.__prodi = prodi

    # Get prodi
    def getProdi(self):
        return self.__prodi

    # set kode
    def setKode(self, kode):
        self.__kode = kode

    # Get kode
    def getKode(self):
        return self.__kode

    # Get list course
    def get_course_list(self):
        return self.course_list

    # method tambah course
    def add_course(self, course):
        self.course_list.append(course)

    # method hapus course
    def remove_course(self, course):
        self.course_list.remove(course)

File: C++/Program_Studi.cpp
#include "header.hh"

// Constructor
Program_Studi::Program_Studi()
{
    prodi = "";
    kode = "";
}

// Constructor with all attributes
Program_Studi::Program_Studi(string prodi, string kode)
{
    this->prodi = prodi;
    this->kode = kode;
}

// Setter and Getter methods for private attributes
// Setter method for prodi
void Program_Studi::setProdi(string prodi)
{
    this->prodi = prodi;
}

// Getter method for prodi
string Program_Studi::getProdi()
{
    return this->prodi;
}

// Setter method for kode
void Program_Studi::setKode(string kode)
{
    this->kode = kode;
}

// Getter method for kode
string Program_Studi::getKode()
{
    return this->kode;
}

// getter method for course_list
vector<Course> Program_Studi::get_course_list()
{
    return course_list;
}

//  method for add course
void Program_Studi::add_course(Course course)
{
    course_list.push_back(course);
}
// Destructor
Program_Studi::~Program_Studi()
{
}
